improver_tests/calibration/utilities/test_utilities.py

```diff
@@ -572,5 +578,172 @@ def test_input_not_masked(self):
         self.assertEqual(self.percentiles_land.data.dtype, np.float32)
 
 
+class Test_time_coords_match(IrisTest):
+
+    """Test for function that tests if forecast period and the hour of the
+     forecast_reference_time coordinate match between two cubes."""
+
+    def setUp(self):
+        """Set-up testing."""
+        self.data = np.ones((3, 3), dtype=np.float32)
+        self.ref_cube = set_up_variable_cube(
+            self.data,
+            frt=datetime.datetime(2017, 11, 10, 1, 0),
+            time=datetime.datetime(2017, 11, 10, 4, 0),
+        )
+        coord_spec = TIME_COORDS["time"]
+        bounds = []
+        for bound in [
+            datetime.datetime(2017, 11, 10, 0, 0),
+            datetime.datetime(2017, 11, 10, 1, 0),
+        ]:
+            bounds.append(date2num(bound, coord_spec.units, coord_spec.calendar))
+
+        self.ref_cube.coord("forecast_reference_time").bounds = bounds
+        self.message = "The following coordinates of the two cubes do not match"
+
+    def test_match(self):
+        """Test returns None when cubes time coordinates match."""
+        result = time_coords_match(self.ref_cube, self.ref_cube.copy())
+        self.assertIsNone(result)
+
+    def test_forecast_period_mismatch(self):
+        """Test an error is raised when the forecast period mismatches."""
+        self.adjusted_cube = set_up_variable_cube(
+            self.data,
+            frt=datetime.datetime(2017, 11, 10, 1, 0),
+            time=datetime.datetime(2017, 11, 10, 5, 0),
+        )
+
+        with self.assertRaisesRegex(ValueError, self.message):
+            time_coords_match(self.ref_cube, self.adjusted_cube)
+
+    def test_frt_hour_without_bounds_mismatch(self):
+        """Test an error is raised when the forecast_reference_time mismatches"""
+        self.adjusted_cube = set_up_variable_cube(
+            self.data,
+            frt=datetime.datetime(2017, 11, 10, 2, 0),
+            time=datetime.datetime(2017, 11, 10, 5, 0),
+        )
+
+        with self.assertRaisesRegex(ValueError, self.message):
+            time_coords_match(self.ref_cube, self.adjusted_cube)
+
+    def test_frt_hour_with_bounds_mismatch(self):
+        """Test returns None when the forecast_reference_time bounds are
+        absent from one cube"""
+        self.adjusted_cube = self.ref_cube.copy()
+        self.adjusted_cube.coord("forecast_reference_time").bounds = None
+        result = time_coords_match(self.ref_cube, self.adjusted_cube)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Assigning result of a function call, where the function has no return](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5439763)